sycl/include/CL/sycl/detail/array.hpp

```diff
@@ -20,8 +20,28 @@ namespace detail {
 
 template <int dimensions = 1> class array {
   static_assert(dimensions >= 1, "Array cannot be 0-dimensional.");
+
 public:
-  array() : common_array{0} {}
+  template <int N = dimensions,
```

Comment:
Please, add a TODO here with the text from here: https://github.com/intel/llvm/pull/359#issuecomment-514629489